config/jobs/kubernetes-csi/csi-driver-host-path/csi-driver-host-path-config.yaml

```diff
@@ -0,0 +1,606 @@
+      preset-dind-enabled: "true"
+      preset-bazel-remote-cache-enabled: "true"
+      preset-kind-volume-mounts: "true"
+    spec:
+      containers:
+      # We need this image because it has Docker in Docker and go.
+      - image: gcr.io/k8s-testimages/kubekins-e2e:v20190329-811f7954b-master
+        command:
+        - runner.sh
+        args:
+        - ./.prow.sh
+        env:
+        - name: CSI_PROW_KUBERNETES_VERSION
+          value: "latest"
+        # docker-in-docker needs privileged mode
+        securityContext:
+          privileged: true
+      resources:
+        requests:
+          # these are both a bit below peak usage during build
+          # this is mostly for building kubernetes
+          memory: "9000Mi"
+          # during the tests more like 3-20m is used
+          cpu: 2000m
+
+periodics:
+- interval: 6h
+  name: ci-kubernetes-csi-1-13-on-kubernetes-1-13
+  decorate: true
+  extra_refs:
+  # TODO: replace with kubernetes-csi/csi-driver-host-path master once branch prow is merged
+  - org: pohly
+    repo: csi-driver-host-path
+    base_ref: prow
+  labels:
+    preset-service-account: "true"
+    preset-dind-enabled: "true"
+    preset-bazel-remote-cache-enabled: "false"
+    preset-kind-volume-mounts: "true"
+  spec:
+    containers:
+    # We need this image because it has Docker in Docker and go.
+    - image: gcr.io/k8s-testimages/kubekins-e2e:v20190329-811f7954b-master
+      command:
+      - runner.sh
+      args:
+      - ./.prow.sh
+      env:
+      - name: CSI_PROW_KUBERNETES_VERSION
+        value: "release-1.13"
+      - name: CSI_PROW_BUILD_JOB
+        value: "false"
+      - name: CSI_PROW_DEPLOYMENT
+        value: "kubernetes-1.13"
+      # docker-in-docker needs privileged mode
+      securityContext:
+        privileged: true
+        resources:
+          requests:
+            cpu: 2000m
+- interval: 6h
+  name: ci-kubernetes-csi-1-13-on-kubernetes-1-14
+  decorate: true
+  extra_refs:
+  # TODO: replace with kubernetes-csi/csi-driver-host-path master once branch prow is merged
+  - org: pohly
+    repo: csi-driver-host-path
+    base_ref: prow
+  labels:
+    preset-service-account: "true"
+    preset-dind-enabled: "true"
+    preset-bazel-remote-cache-enabled: "false"
+    preset-kind-volume-mounts: "true"
+  spec:
+    containers:
+    # We need this image because it has Docker in Docker and go.
+    - image: gcr.io/k8s-testimages/kubekins-e2e:v20190329-811f7954b-master
+      command:
+      - runner.sh
+      args:
+      - ./.prow.sh
+      env:
+      - name: CSI_PROW_KUBERNETES_VERSION
+        value: "release-1.14"
+      - name: CSI_PROW_BUILD_JOB
+        value: "false"
+      - name: CSI_PROW_DEPLOYMENT
+        value: "kubernetes-1.13"
+      # docker-in-docker needs privileged mode
+      securityContext:
+        privileged: true
+        resources:
+          requests:
+            cpu: 2000m
+- interval: 6h
+  name: ci-kubernetes-csi-1-13-on-kubernetes-master
```

Comment:
To reduce our testing matrix, I think we should only test "-1, current, +1" version skew with K8s.

So we would have the following jobs:
* driver-1.13-on-kubernetes-1.13
* driver-1.13-on-kubernetes-1.14

* driver-1.14-on-kubernetes-1.13
* driver-1.14-on-kubernetes-1.14
* driver-1.14-on-kubernetes-master

* driver-master-on-kubernetes-1.14
* driver-master-on-kubernetes-master

Reply:
But we still promise that "1.13 on Kubernetes 1.15" works, right? How can we promise that if we don't test it?

Reply:
Ok, I can see the point about 1.13 on k8s 1.15. 

But thinking about the newer configs, I think we are misinterpreting the version skew support.  Let's say 1.14 adds some new features that were not present in 1.13 and the driver wants to use them.  Then the 1.14 deployment enables those new features and cannot be backwards compatible because we only offer backwards compatibility if you don't use the new features.  So the deployment should never be used on an older K8s, and the jobs look like:

* driver-1.13-on-k8s-1.13 
* driver-1.13-on-k8s-1.14
* driver-1.13-on-k8s-master

* driver-1.14-on-k8s-1.14
* driver-1.14-on-k8s-master

* driver-master-on-k8s-master

Reply:
> Let's say 1.14 adds some new features that were not present in 1.13 and the driver wants to use them. Then the 1.14 deployment enables those new features and cannot be backwards compatible because we only offer backwards compatibility if you don't use the new features. 

Correct, if 1.14 enables and depends on new features, then it cannot run on older Kubernetes. But we still may want to test that the new sidecars with the new features disabled work on Kubernetes 1.13. It'll depend on the feature whether that can work and is supported: if it doesn't, we announce a breaking change and release a new major version.

So for the upcoming sidecars we theoretically have two deployments: with topology enabled and disabled. The one where it's off can run on 1.13, and this is something that we have to support and test - at least that was my understanding of the compatibility document, and the new versions (1.1 instead of 2.0) seems to confirm that.

Reply:
I think having different deployments with different feature sets in the same release is going to be too complex to maintain, and have too many permutations to test.

Let's just start with just like-version testing for now and iron out the details for skew testing later.

Reply:
Right now the situation is simple: we have one deployment for each release and and we can test version skew. Can we merge this as-is to test that the scripts work and then revise later?

Reply:
If you want test newer sidecars on older deployments, you could still use the 1.13 deployment on K8s 1.13 job to do that.  If run in the sidecar pull, it will use the built image.  We could also have a CI job that uses canary images instead of the stable images.  Still that doubles the amount of jobs we have to 12.

What if we don't have a new hostpath deployment for every release, and only add a new deployment when something in the hostpath deployment actually changes?

So that would leave us with only 6 jobs in CI:
Hostpath with stable images * K8s 1.13, 1.14, master
Hostpath with canary images * K8s 1.13, 1.14, master

cc @jsafrane for any opinions

Reply:
I think testing a newer deployment, which might have enabled new features against older Kubernetes is not correct, and is not something we support in our versioning policy.

Reply:
(a newer sidecar using an older deployment is something we do support)

Reply:
> If you want test newer sidecars on older deployments, you could still use the 1.13 deployment on K8s 1.13 job to do that.

Testing "canary" images with older deployments is something that can be done because it is unambiguous what those images are. It gets more tricky when we have multiple release trains: "stable images" becomes ambiguous in that case. Which ones do we test on 1.13, and in which combination? There has to be a definition of that somewhere. Currently the example deployments are our definition of which sidecars are meant to be used together for a certain situation.

> So that would leave us with only 6 jobs in CI:
> Hostpath with stable images * K8s 1.13, 1.14, master
> Hostpath with canary images * K8s 1.13, 1.14, master

Which "stable images"?

> What if we don't have a new hostpath deployment for every release, and only add a new deployment when something in the hostpath deployment actually changes?

What is this "something"? Does adding a new release train count? Because that is what changes now: we support external-provisioner 1.0.x for Kubernetes 1.13 and we add 1.1.x for Kubernetes 1.14. Nothing else changes in the deployment.

If a new release fully replaces the previous one, we can just bump the image version in the existing deployment. But this is not the case right now.

> I think testing a newer deployment, which might have enabled new features against older Kubernetes is not correct, and is not something we support in our versioning policy.

I agree, and this is not what I was suggesting. I was talking about a new deployment that does not enable new features.

Let's take the tentative "kubernetes-1.14" deployment as example. Assume that we do the 1.1 releases and then update that deployment with those new image versions. We don't enable any new features. Do you want this deployment to be in the csi-driver-host-path repo? What kind of testing do you want to do with that deployment?

cc @jsafrane

Reply:
In order to make some progress with this PR and perhaps get it merged, I've taken out the `ci-kubernetes-csi-1-14-on-*` jobs.

@msau42 is that now something that could be merged?

Reply:
In order to test older, but still supported release trains, I think we will need to branch every release and have a job off of the release branch that tests the supported versions at the time of release.

There's a lot of detail and logistics that I think we still need to work out, so I am fine with starting with a smaller test matrix until we can document and agree on a plan.